extensions/resource-deployment/src/services/resourceTypeService.ts

```diff
@@ -370,3 +350,30 @@ async function exists(path: string): Promise<boolean> {
 		return false;
 	}
 }
+
+/**
+ * processWhenClause takes in a when clause (either the word 'true' or a series of clauses in the format:
+ * '<type_name>=<value_name>' joined by '&&').
+ * If the when clause is true or undefined, return true as there is no clause to check.
+ * It evaluates each individual when clause by comparing the equivalent selected options (sorted in alphabetical order and formatted to match).
+ * If there is any selected option that doesn't match, return false.
+ * Return true if all clauses match.
+ */
+export function processWhenClause(when: string, selectedOptions: { option: string, value: string }[]): boolean {
+	if (when === undefined || when.toString().toLowerCase() === 'true') {
+		return true;
+	} else {
+		const expected = when.replace(/\s/g, '').split('&&').sort();
+		let actual: string[] = [];
```

Comment:
Just simplify this to ``const actual = selectedOptions.map(option => `${option.option}=${option.value}`);``